tests/util/pool_tests.rs

```diff
@@ -127,4 +130,17 @@ pub fn test_thinpool_thindev_destroy(paths: &[&Path]) -> () {
                                  Sectors(128u64));
     assert!(thindev.is_err());
     pool.teardown().unwrap();
+
+    // Check that destroyed fs is not present in MDV. If the record
+    // had been left on the MDV that didn't match a thin_id in the
+    // thinpool, ::setup() will fail.
+    let paths2 = paths
+        .into_iter()
+        .map(|x| x.to_path_buf())
+        .collect::<Vec<_>>();
+    let mut pool2 = StratPool::setup(pool_uuid, &paths2).unwrap();
+
+    // This also should never happen, given the previous two parts of
+    // this test.
+    assert!(pool2.get_filesystem(&fs_uuid).is_none())
```

Comment:
Unfortunately, are tests have to clean up after themselves, including doing a teardown of any pools that were setup. Right now this is causing the subsequent test too fail.